http-server/src/lib.rs

```diff
@@ -28,10 +28,40 @@ mod access_control;
 mod response;
 mod server;
 
+use std::fmt::{self, Debug, Display};
+
 pub use access_control::{AccessControl, AccessControlBuilder, AllowHosts, Host};
 pub use jsonrpsee_types::{Error, TEN_MB_SIZE_BYTES};
 pub use jsonrpsee_utils::server::rpc_module::{Methods, RpcContextModule, RpcModule};
 pub use server::{Builder as HttpServerBuilder, Server as HttpServer};
 
 #[cfg(test)]
 mod tests;
+
+#[derive(Debug, thiserror::Error)]
+/// Http-specific error type.
+pub struct HttpError(Error);
+
+impl Display for HttpError {
+	fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+		Display::fmt(&self.0, f)
+	}
+}
+
+impl From<hyper::Error> for HttpError {
+	fn from(hyper_err: hyper::Error) -> HttpError {
+		HttpError(Error::Transport(Box::new(hyper_err)))
+	}
+}
+
+impl From<std::io::Error> for HttpError {
```

Comment:
is not `From<std::io::Error> for Error` in jsonrpsee_types sufficient?

Reply:
It is, but there's an `io::Error` raised in `ws-server` as well. Maybe I can move the conversion from `jsonrpsee_types` to `ws-server` (but it has to be in two places either way).

The downside of this PR is that it becomes less obvious where errors are defined/handled. I'm not sure if I like it tbh.